src/Symfony/Component/Webhook/Tests/Server/JsonBodyConfiguratorTest.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Webhook\Tests\Server;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpClient\HttpOptions;
+use Symfony\Component\RemoteEvent\RemoteEvent;
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Component\Webhook\Server\JsonBodyConfigurator;
+use Symfony\Component\Webhook\Server\PayloadEncoderInterface;
+
+class JsonBodyConfiguratorTest extends TestCase
+{
+    public function testPayloadWithPayloadEncoder()
+    {
+        $payload = ['foo' => 'bar'];
+
+        $payloadEncoder = $this->createMock(PayloadEncoderInterface::class);
+        $payloadEncoder
+            ->expects($this->once())
+            ->method('encode')
+            ->with($payload)
+        ;
+
+        $httpOptions = new HttpOptions();
+        $httpOptions->setHeaders([
+            'Webhook-Event' => 'event-name',
+            'Webhook-Id' => 'event-id',
+        ]);
+
+        $configurator = new JsonBodyConfigurator($payloadEncoder);
+        $configurator->configure(new RemoteEvent('event-name', 'event-id', $payload), 's3cr3t', $httpOptions);
+    }
+
+    public function testPayloadWithSerializer()
```

Comment:
Shall we test that the encoded payload returned by the serializer is actually used?